pkg/operator/controller/ingress/controller.go

```diff
@@ -1271,13 +1279,47 @@ func (r *reconciler) allRouterPodsDeleted(ingress *operatorv1.IngressController)
 	return true, nil
 }
 
-// computeUpdatedInfraFromService updates GCP's PlatformStatus with Ingress LB IPs when the DNSType is `ClusterHosted`.
+// computeUpdatedInfraFromService updates PlatformStatus for GCP and AWS with Ingress LB IPs when the DNSType is `ClusterHosted`.
 func computeUpdatedInfraFromService(service *corev1.Service, infraConfig *configv1.Infrastructure) (bool, error) {
 	platformStatus := infraConfig.Status.PlatformStatus
 	if platformStatus == nil {
 		return false, fmt.Errorf("invalid PlatformStatus within Infrastructure config")
 	}
+	ipCmpOpts := []cmp.Option{
+		cmpopts.EquateEmpty(),
+		cmpopts.SortSlices(func(a, b configv1.IP) bool {
+			return a < b
+		}),
+	}
 	switch platformStatus.Type {
+	case configv1.AWSPlatformType:
+		if platformStatus.AWS != nil && platformStatus.AWS.CloudLoadBalancerConfig != nil && platformStatus.AWS.CloudLoadBalancerConfig.DNSType == configv1.ClusterHostedDNSType {
+			// The cluster has to run its own CoreDNS pod for DNS. Update Infra CR
+			// with the Ingress LB IPs. These values are used to configure the
+			// in-cluster DNS to provide resolution for *.apps.
+			if platformStatus.AWS.CloudLoadBalancerConfig.ClusterHosted == nil {
+				platformStatus.AWS.CloudLoadBalancerConfig.ClusterHosted = &configv1.CloudLoadBalancerIPs{}
+			}
+			ingresses := service.Status.LoadBalancer.Ingress
+			ingressLBIPs := []configv1.IP{}
+			for _, ingress := range ingresses {
+				ingressIPs, err := net.LookupIP(ingress.Hostname)
```

Comment:
The AWS Load Balancer's IPs change frequently. Any time the router pods rollout, nodes scale up/down, or internal AWS maintenance. From what I saw, AWS recommends against using LB hostname's resolved IPs, as they can go stale for a number of reasons, and are not guaranteed to persist. The TTL is 60 seconds, so they could change as often as every 60 seconds, but realistically it's not that often. Even if the downstream components handle these updates, there could still be buggy behavior with gaps between IP changes  and CIO reconcilation.

Could you help me understand why we chose IPs over hostname for AWS in the infra config? If the downstream product is going to be CoreDNS serving this *.api wildcard, then can't CoreDNS serve the hostname as a CNAME record?

Reply:
@gcs278 We are aware of this limitation that the LB IP address could change during the life of the cluster. During the time of investigation, CoreDNS could not handle CNAME records. There are some untested workarounds suggested. Until we figure out a way to directly use the LB Hostname, in-cluster DNS for AWS in this form (using the IP corresponding to the LB Hostname), would be in tech-preview.

Reply:
> CoreDNS could not handle CNAME records.

Hmm yea, I see now after some testing of the CoreDNS file plugin. Regardless, the IP address needs to be stored because the CNAME is just another hostname, which needs to also be looked up. Darn, that's tricky. You might need to force the use of static EIPs to solve this (I assume you've also explored this).

> Until we figure out a way to directly use the LB Hostname, in-cluster DNS for AWS in this form (using the IP corresponding to the LB Hostname), would be in tech-preview.

So to be clear, we are going to try to figure this out for GA? And this will *hopefully* change? If so, do you mind putting a comment regard that here? We should make it clear that this is a known problem:

```suggestion
                                // Resolving the LoadBalancer's IPs is not ideal because they may change, but currently there is no better alternative.
				ingressIPs, err := net.LookupIP(ingress.Hostname)
```

Reply:
@gcs278 yes, this feature is definitely staying in TP until we figure out a better alternative. Proceeding in TP with this workaround is still useful. I still have some unanswered questions around using static EIPs. Happy to have those discussions to get to a better solution.
Yes, suggestion definitely helps.